source/uwp/SharedRenderer/lib/AdaptiveImageRenderer.cpp

```diff
@@ -367,314 +371,219 @@ namespace AdaptiveCards::Rendering::Xaml_Rendering
+                strongThis->m_copyStreamOperations.push_back(copyStreamOperation);
+                co_await copyStreamOperation;
+                co_return randomAccessStream;
             }
-
-            SetImageSource(imgProperties.uiElement, image, imgProperties.stretch);
-
-            // Issue #8126
-            if (imgProperties.isAutoSize)
-            {
-                SetAutoSize(imgProperties.uiElement,
-                            imgProperties.parentElement,
-                            imgProperties.imageContainer,
-                            imgProperties.isVisible,
-                            true /* imageFiresOpenEvent */);
-            }
-
-            return image;
-        }
-        else
-        {
-            return PopulateImageFromUrlAsync(imageUrl, imgProperties);
         }
-    }
 
-    winrt::ImageSource XamlBuilder::CreateImageSource(bool isImageSvg)
-    {
-        if (isImageSvg)
-        {
-            winrt::SvgImageSource svgImageSource{};
-            return svgImageSource;
-        }
-        else
-        {
-            winrt::BitmapImage bitmapImage{};
-            return bitmapImage;
-        }
+        co_return {};
     }
 
-    // Issue #8127
     template<typename TElement>
-    winrt::ImageSource XamlBuilder::PopulateImageFromUrlAsync(winrt::Uri const& imageUrl,
-                                                              ImageProperties<TElement> const& imgProperties)
+    winrt::fire_and_forget XamlBuilder::ResolveImageAsync(winrt::Uri const uri,
+        winrt::AdaptiveCardResourceResolvers const resolvers,
+        winrt::ImageSource imageSource,
+        ImageProperties<TElement> const properties)
     {
-        winrt::HttpBaseProtocolFilter httpBaseProtocolFilter{};
-        httpBaseProtocolFilter.AllowUI(false);
-
-        winrt::HttpClient httpClient{httpBaseProtocolFilter};
+        auto isAutoSize{properties.isAutoSize};
+        auto isSVG{properties.isImageSvg};
+        auto weakThis{get_weak()};
+        auto weakImageSource{winrt::make_weak(imageSource)};
+        auto weakFrameworkElement = winrt::make_weak(properties.uiElement);
 
-        // Create an ImageSource to hold the image data.  We use BitmapImage & SvgImageSource in order to allow
-        // the tracker to subscribe to the ImageLoaded/Failed events
-        auto image = CreateImageSource(imgProperties.isImageSvg);
-        this->m_imageLoadTracker->TrackImage(image);
-
-        if (!imgProperties.isImageSvg)
+        try
         {
-            image.as<winrt::BitmapImage>().CreateOptions(winrt::BitmapCreateOptions::None);
-        }
-
-        auto getStreamOperation = httpClient.GetInputStreamAsync(imageUrl);
-        getStreamOperation.Completed(
-            [this, weakThis = this->get_weak(), imgProperties, image](
-                auto const& operation, auto status) -> void
+            // if image loading is handled by Xaml, we set this to true;
+            auto imageFiresOpenEvent{true};
```

Comment:
Just confirming my understanding:

`imageFiresOpenEvent` is only used by Xaml handling. We default the value to `true` and set it to `false` if `ResolveToStringAsync` returns a stream. `ResolveToStreamAsync` will simply return a nullptr if we did not resolve the stream